### ISSUE TITLE
Connection close directly when the last get param value is empty.

### ISSUE THREAD

**Issue by cuber** [ellzey/libevhtp#167]
Environment `Centos 5.8` with `libevhtp-1.2.10` `libevent-2.0.21-stable` `openssl-1.0.1j`.
I am just using the helloworld example below.
```c
#include <stdio.h>
#include <evhtp.h>

void
testcb(evhtp_request_t * req, void * a) {
    evbuffer_add_reference(req->buffer_out, "foobar", 6, NULL, NULL);
    evhtp_send_reply(req, EVHTP_RES_OK);
}

int
main(int argc, char ** argv) {
    evbase_t * evbase = event_base_new();
    evhtp_t  * htp    = evhtp_new(evbase, NULL);

    evhtp_set_cb(htp, "/test", testcb, NULL);
    evhtp_bind_socket(htp, "0.0.0.0", 8082, 1024);
    event_base_loop(evbase, 0);
    return 0;
}
```
Complie command 
```bash
gcc evhtp.c -o evhtp -levhtp \
  -I/path/to/libevhtp-1.2.10/include \
  -I/path/to/libevent-2.0.21-stable/include \
  -L/path/to/libevhtp-1.2.10/lib \
  -L/path/to/libevent-2.0.21-stable/lib \
  -L/path/to/openssl-1.0.1j/lib
````
When the _last_ GET param value is empty, connection close directly without response.
```bash
curl "http://127.0.0.1:8082/test?abc=1&def=" -v
* About to connect() to 127.0.0.1 port 8082
*   Trying 127.0.0.1... connected
* Connected to 127.0.0.1 (127.0.0.1) port 8082
> GET /test?abc= HTTP/1.1
> User-Agent: curl/7.15.5 (x86_64-redhat-linux-gnu) libcurl/7.15.5 OpenSSL/0.9.8b zlib/1.2.3 libidn/0.6.5
> Host: 127.0.0.1:8082
> Accept: */*
>
* Empty reply from server
* Connection #0 to host 127.0.0.1 left intact
curl: (52) Empty reply from server
* Closing connection #0
```

**Comment by ellzey** [comment]
whoah, that's scary, this is a result of a set of patches that tried to make argument parsing more "strict"; but there are flags you can toggle to adjust how strict.

Try `evhtp_set_parser_flags(htp, EVHTP_PARSE_QUERY_FLAG_LENIENT);` 

I should probably make this the default.

**Comment by ellzey** [comment]
bump

**Comment by tjk** [comment]
@ellzey where do you want to go with this? I kind of thinking (regardless of query parsing "flag") that we should not error out in `_evhtp_request_parser_args` (causing `htparse` to fail) if `uri->query` is `NULL` (but that might involve a lot of refactoring if `uri->query` expects to be non-`NULL`).

**Comment by ellzey** [comment]
@tjk I think we went the wrong way with parser flags. Instead of OFF's we should have had ON's. The initial parser followed the wikipedia formula for arguments; the extra stuff is throwing things off.

**Comment by ellzey** [comment]
@tjk I just made EVHTP_PARSE_QUERY_FLAG_LENIENT the default. Like, just now.

**Comment by cuber** [comment]
Thanks @ellzey the latest develop fixed this problem.

**Comment by jacobmarble** [comment]
I'm bumping into this problem with 1.2.10; upgrading to 1.2.11n fixes it. However, I need to support older versions of libevhtp.

Per @ellzey's suggestion, I tried adding `evhtp_set_parser_flags(htp, EVHTP_PARSE_QUERY_FLAG_LENIENT);`, but that doesn't help. When should this be set? I have tried immediately after `evhtp_new`, and also after `evhtp_bind_socket` (the first two evhtp calls my code makes).
